articles/quickstart/native/ionic4/01-login.md

```diff
@@ -0,0 +1,443 @@
+    this.storage.remove('expires_at');
+    this.accessToken = null;
+    this.user = null;
+    this.loggedIn = false;
+
+    this.safariViewController.isAvailable()
+      .then((available: boolean) => {
+        const domain = AUTH_CONFIG.domain;
+        const clientId = AUTH_CONFIG.clientId;
+        const pkgId = AUTH_CONFIG.packageIdentifier;
+        let url = `https://<%= "${domain}" %>/v2/logout?client_id=<%= "${clientId}" %>&returnTo=<%= "${pkgId}" %>://<%= "${domain}" %>/cordova/<%= "${pkgId}" %>/callback`;
+        if (available) {
+          this.safariViewController.show({
+            url: url
+          })
+          .subscribe((result: any) => {
+              if(result.event === 'opened') console.log('Opened');
+              else if(result.event === 'loaded') console.log('Loaded');
+              else if(result.event === 'closed') console.log('Closed');
+            },
+            (error: any) => console.error(error)
+          );
+        } else {
+          // use fallback browser
+          cordova.InAppBrowser.open(url, '_system');
+        }
+      }
+    );
+  }
+}
+```
+
+## Add Authentication with Auth0
+
+Add a control to your app to allow users to log in. The control should call the `login` method from the `AuthService`. Start by injecting the `AuthService` in a component.
+
+```js
+// src/app/home/home.page.ts
+import { Component } from '@angular/core';
+import { AuthService } from '../../services/auth.service';
+
+@Component({
+  selector: 'app-home',
+  templateUrl: 'home.page.html',
+  styleUrls: ['home.page.scss']
+})
+export class HomePage {
+
+  constructor(
+    public auth: AuthService
+  ) {}
+
+}
+```
+
+The `AuthService` is now accessible in the view and its `login` method can be called.
+
+```html
+<!-- src/app/home/home.page.html -->
+
+<ion-header>
+  <ion-toolbar>
+    <ion-title>
+      Home
+    </ion-title>
+  </ion-toolbar>
+</ion-header>
+
+<ion-content padding>
+  <p *ngIf="auth.loading" text-center>Loading...</p>
+  <ng-template [ngIf]="!auth.loading">
+    <!-- Not loading, not logged in: show login button -->
+    <ion-button
+      button
+      expand="block"
+      color="primary"
+      *ngIf="!auth.loggedIn"
+      (click)="auth.login()">Log In</ion-button>
+    <!-- Not loading, logged in: show profile and logout button -->
+    <ng-template [ngIf]="auth.loggedIn">
+      <ion-card *ngIf="auth.user">
+        <img [src]="auth.user.picture">
+        <ion-card-content>
+          <ion-card-title>{{ auth.user.name }}</ion-card-title>
+          <pre>{{ auth.user | json }}</pre>
+        </ion-card-content>
+      </ion-card>
+      <ion-button
+        button
+        expand="block"
+        color="danger"
+        (click)="auth.logout()">Log Out</ion-button>
+    </ng-template>
+  </ng-template>
+</ion-content>
+```
+
+### Update Auth0 Dashboard Configuration
+
+1. After executing the `run` command, your `config.xml` file should contain `<allow-navigation>` tag(s).
```

Comment:
I didn't have this in my `config.xml` after following all the steps

Reply:
I think the section `Update Auth0 Dashboard Configuration` could be removed. This steps are only needed when changing the default config for `Ionic Web View`, but neither this tutorial and the sample are doing it.